src/widgetastic/browser.py

```diff
@@ -366,6 +370,12 @@ def click(self, locator, *args, **kwargs):
         if not ignore_ajax:
             try:
                 self.plugin.ensure_page_safe()
+            except TimedOutError:
+                try:
+                    self.plugin.after_click_safe_timeout(el, locator)
+                except UnexpectedAlertPresentException:
+                    pass
+                raise
```

Comment:
`UnexpectedAlertPresentException` will be reraised here.

Reply:
So put it into `finally` maybe?

Reply:
```python
try:
    self.plugin.after_click_safe_timeout(el, locator)
except UnexpectedAlertPresentException:
    pass
except Exception:
    raise
```